File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonTypeDisableCommand.cs
using SIE.Andon.Andons;
using SIE.Web.Command;
using System.Linq;

namespace SIE.Web.Andon.Andons.Commands
{
    /// <summary>
    /// 禁用安灯类型维护
    /// </summary>
    public class AndonTypeDisableCommand : ViewCommand<double[]>
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(double[] args, string scope)
        {
            RT.Service.Resolve<AndonTypeController>().DisableAndonType(args.ToList());
            return true;
        }
    }
}

File: Modules/SMES/Common/SIE.Web.Andon/Andons/Commands/AndonTypeDisableCommand.js
SIE.defineCommand('SIE.Web.Andon.Andons.Commands.AndonTypeDisableCommand', {
    meta: { text: "禁用", group: "business", iconCls: "icon-NetworkError icon-red" },
    canExecute: function (listView) {
        var selectModels = listView.getSelection();
        if (selectModels.length == 0) return false;
        var res = true;
        SIE.each(selectModels, function (model) {
            if (model.data.State == 0)
                res = false;
        });
        return res;
    },
    execute: function (listView, source) {
        SIE.Msg.askQuestion('确认禁用选中的安灯类型？'.t(), function () {
            listView.execute({
                data: listView.getSelectionIds(),
                success: function (res) {
                    SIE.each(listView.getSelection(), function (model) {
                        model.data.State = 0;
                    });
                    listView.reloadData();
                }
            });
        });
    }
});
